Sources/Meow/Model.swift

```diff
@@ -152,8 +152,10 @@ extension MutableModel {
     @discardableResult
     public func save(in database: MeowDatabase) async throws -> MeowOperationResult {
         let reply = try await database.collection(for: Self.self).upsert(self)
+        let updateSuccess = reply.updatedCount == 1 && reply.upserted == nil
+        let upsertSuccess = reply.updatedCount == 0 && reply.upserted != nil
```

Comment:
I don't feel comfortable wit these conditions. I know it can't happen, but this code would fail when `updatedCount > 1` or `reply.updatedCount != 0 && reply.upserted != nil`. It feels very error prone, and we're really only interested in the existance of this insert. It would probably be simpler to read if we said `reply.updatedCount > 0 || reply.upserted != nil`